File: lc_soln/0003_longest_substring_without_repeating_characters/solution.cs
public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        int result = 0;
        int start = 0;
        var counter = new Dictionary<char, int>();
        for (int i = 0; i < s.Length; ++i)
        {
            while (counter.TryGetValue(s[i], out int val) && val > 0)
            {
                counter[s[start]]--;
                start++;
            }
            counter[s[i]] = counter.GetValueOrDefault(s[i]) + 1;
            result = int.Max(result, i - start + 1);
        }
        return result;
    }
}

File: lc_soln/0003_longest_substring_without_repeating_characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        result = 0
        start = 0
        counter = {}
        for i, c in enumerate(s):
            while counter.get(c, 0) > 0:
                counter[s[start]] -= 1
                start += 1
            counter[c] = counter.setdefault(c, 0) + 1
            result = max(result, i - start + 1)
        return result


def main(): ...


if __name__ == "__main__":
    main()
